lib/features/personal_summary/yearly_overview/cubit/yearly_overview_state.dart

```diff
@@ -113,7 +116,8 @@ final class YearlyOverviewState extends Equatable {
       );
     }
 
-    for (var it in mergedList) {
+    final copyOfMergedList = mergedList;
```

Comment:
If I'm correct, this doesn't copy a list, but will make a reference to the other list.

If you want to copy I think you need to do:
Copylist = [...mergedlist]

And even that is not fully a copy. It's only creating a new list with the same objects (references to those objects)

So depending on what you need you can keep it or you need to change it :)